sharder/deterministic.go

```diff
@@ -117,33 +118,54 @@ func (d *DeterministicSharder) Start() error {
 		}
 		d.Logger.Debug().Logf("picked up local peer port of %s", localPort)
 
-		// get my local interfaces
+		// get my local interfaces' IPs.
 		localAddrs, err := net.InterfaceAddrs()
 		if err != nil {
 			return errors.Wrap(err, "failed to get local interface list to initialize sharder")
 		}
+		localIPs := make([]string, len(localAddrs))
+		for i, addr := range localAddrs {
+			addrStr := addr.String()
+			ip, _, err := net.ParseCIDR(addrStr)
+			if err != nil {
+				return errors.Wrap(err, fmt.Sprintf("failed to parse CIDR for local IP %s", addrStr))
+			}
+			localIPs[i] = ip.String()
+		}
+
+		// If RedisIdentifier is an IP, add it to localIps.
+		if redisIdentifier, err := d.Config.GetRedisIdentifier(); err == nil && redisIdentifier != "" {
```

Comment:
Yes, as can be seen in `redis.go`, if `RedisIdentifier` is specified then it will be the peer address shown, and you can safely replace the entire contents of the list with the single item. Please do that. 

I'd also look favorably on inverting the logic within the publicAddr function so that if redisIdentifier is specified, it doesn't go through all of the other logic. If you felt up to adding that here, it would make sense. Let me know if not.

Reply:
Done, both parts.